edb/pgsql/resolver/command.py

```diff
@@ -2315,17 +2396,82 @@ def _resolve_conflict_update_rel(
             ),
         ))
 
-    if cu_rel.ctes:
-        ctx.ctes_buffer.extend(cu_rel.ctes)
-        cu_rel.ctes = None
+        # inject subject id from "else" rvar
+        subject_id_col = _get_path_id_output(
+            else_cte.query, compiled_dml.subject_id, compiled_dml,
+            aspect=pgce.PathAspect.IDENTITY
+        )
+        assert isinstance(subject_id_col, pgast.ColumnRef)
+        cu_rel.target_list.append(pgast.ResTarget(
+            val=pgast.ColumnRef(
+                name=('else', subject_id_col.name[-1]),
+            ),
+            name='id'
+        ))
+
+        # add a join condition for "excluded" and "subject" rvars
 
-    cu_table.alias = ctx.alias_generator.get('rel')
+        # We start with a plain value_id, but because of for loops, rvar_map
+        # will contain path_ids polluted with namespaces. So instead of a plain
+        # one, we find a path_id from rvar_map that matches the plain one in all
+        # but the namespace.
+        value_id = next(
+            p for p, _ in else_cte.query.path_rvar_map.keys()
+            if p.replace_namespace(set()) == compiled_dml.value_id
+        )
```

Comment:
Another option would be to use the IR to find it?

Reply:
Yes, that's possible, but:
- we'd have to attach the IR stmt from the middle step (in `_compile_uncompiled_dml`) to `CompiledDML` so it can be used later on during resolving,
- value_id does not appear at a convenient location in IR. It is used only when referencing the value relation to get values for the update shape
![image](https://github.com/user-attachments/assets/61de4a9d-d148-4efc-8cc1-0b76543ef835)

Alternatively, we could just look at namespaces on the subject's path_id and apply those namespaces to the plain value_id.